assets/component-image-with-text.css

```diff
@@ -91,56 +91,58 @@
   z-index: -1;
 }
 
-.image-with-text:not(.image-with-text--overlap) .grid__item:first-child:after {
+.image-with-text:not(.image-with-text--overlap) .image-with-text__media-item:after {
   border-radius: var(--media-radius);
   box-shadow: var(--media-shadow-horizontal-offset)
     var(--media-shadow-vertical-offset)
     var(--media-shadow-blur-radius)
     rgba(var(--color-shadow), var(--media-shadow-opacity));
 }
 
-.image-with-text:not(.image-with-text--overlap) .grid__item:last-child:after {
+.image-with-text:not(.image-with-text--overlap) .image-with-text__text-item:after {
   border-radius: var(--text-boxes-radius);
   box-shadow: var(--text-boxes-shadow-horizontal-offset)
     var(--text-boxes-shadow-vertical-offset)
     var(--text-boxes-shadow-blur-radius)
     rgba(var(--color-shadow), var(--text-boxes-shadow-opacity));
 }
 
-.image-with-text .grid__item:first-child > * {
+.image-with-text .image-with-text__media-item > * {
   border-radius: var(--media-radius);
   overflow: hidden;
   box-shadow: var(--media-shadow-horizontal-offset)
-  var(--media-shadow-vertical-offset)
-  var(--media-shadow-blur-radius)
-  rgba(var(--color-shadow), var(--media-shadow-opacity));
+    var(--media-shadow-vertical-offset)
+    var(--media-shadow-blur-radius)
+    rgba(var(--color-shadow), var(--media-shadow-opacity));
 }
 
-.image-with-text .grid__item:last-child > * {
+.image-with-text .global-media-settings {
+  overflow: hidden !important;
+}
+
+.image-with-text .image-with-text__text-item > * {
   border-radius: var(--text-boxes-radius);
   overflow: hidden;
   box-shadow: var(--text-boxes-shadow-horizontal-offset)
-  var(--text-boxes-shadow-vertical-offset)
-  var(--text-boxes-shadow-blur-radius)
-  rgba(var(--color-shadow), var(--text-boxes-shadow-opacity));
-}
-
-.image-with-text:not(.image-with-text--overlap) .grid__item:first-child > * {
-  box-shadow: none;
+    var(--text-boxes-shadow-vertical-offset)
+    var(--text-boxes-shadow-blur-radius)
+    rgba(var(--color-shadow), var(--text-boxes-shadow-opacity));
 }
 
-.image-with-text:not(.image-with-text--overlap) .grid__item:last-child > *{
+.image-with-text:not(.image-with-text--overlap) .image-with-text__media-item > *,
+.image-with-text:not(.image-with-text--overlap) .image-with-text__text-item > * {
   box-shadow: none;
 }
 
 @media screen and (max-width: 749px) {
-  .image-with-text.collapse-corners:not(.image-with-text--overlap) .grid__item:first-child:after,
-  .image-with-text.collapse-corners:not(.image-with-text--overlap) .grid__item .image-with-text__media {
+  .image-with-text.collapse-corners:not(.image-with-text--overlap) .image-with-text__media-item:after,
```

Comment:
Setting the `Border width` to `0` makes the border radius look broken on both Desktop and Mobile.
<img src="https://user-images.githubusercontent.com/37168033/149451176-5ae492b2-7d2c-445e-8b78-174176b86da6.png" width="160"> <img src="https://user-images.githubusercontent.com/37168033/149451510-db09bfa5-46d3-43a1-91a5-80aa508c92d9.png" width="300">

Here's what it looks like if I increase `Border width` to `1`:

<img src="https://user-images.githubusercontent.com/37168033/149451289-5e017f6e-b40c-41fa-b111-e3f71caaf9ac.png" width="160"> <img src="https://user-images.githubusercontent.com/37168033/149451483-b159d468-6123-469a-8720-c17c46d87c99.png" width="300">

Reply:
We do need these styles applied, but I think we might need to revise the `collapse-corners` condition. 
That said, this is already something I can replicate on `main`, so I suggest we create an issue for this, and tackle in a separate PR

<img width="960" alt="Screen Shot 2022-01-14 at 10 22 46 AM" src="https://user-images.githubusercontent.com/26778637/149539936-73916c94-a9da-4399-842e-d63fe294c37c.png">

Reply:
I think that's actually a case we wanted, though Meli and I were going to discuss it more. @melissaperreault this our weird condition right?